query/src/catalogs/impls/meta_backends/embedded_meta_backend.rs

```diff
@@ -1,96 +1,115 @@
-// Copyright 2020 Datafuse Labs.
+//  Copyright 2021 Datafuse Labs.
 //
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
 //
-//     http://www.apache.org/licenses/LICENSE-2.0
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
 //
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
 
 use std::collections::HashMap;
 use std::sync::Arc;
 
 use common_exception::ErrorCode;
-use common_exception::Result;
-use common_flights::meta_api_impl::CreateDatabaseActionResult;
-use common_flights::meta_api_impl::CreateTableActionResult;
-use common_flights::meta_api_impl::DatabaseMetaReply;
-use common_flights::meta_api_impl::DropDatabaseActionResult;
-use common_flights::meta_api_impl::DropTableActionResult;
-use common_flights::meta_api_impl::GetDatabaseActionResult;
-use common_flights::meta_api_impl::GetTableActionResult;
 use common_infallible::RwLock;
 use common_metatypes::MetaId;
 use common_metatypes::MetaVersion;
 use common_planners::CreateDatabasePlan;
 use common_planners::CreateTablePlan;
 use common_planners::DropDatabasePlan;
 use common_planners::DropTablePlan;
-use common_store_api::CommitTableReply;
-use common_store_api::MetaApi;
 
 use crate::catalogs::impls::LOCAL_TBL_ID_BEGIN;
-use crate::catalogs::Database;
-use crate::catalogs::InMemoryMetas;
-use crate::catalogs::MetaBackend;
-use crate::catalogs::TableMeta;
-use crate::datasources::database::local::CsvTable;
-use crate::datasources::database::local::LocalDatabase;
-use crate::datasources::database::local::MemoryTable;
-use crate::datasources::database::local::NullTable;
-use crate::datasources::database::local::ParquetTable;
-use crate::datasources::table::fuse::FuseTable;
+use crate::catalogs::meta_backend::DatabaseInfo;
+use crate::catalogs::meta_backend::MetaBackend;
+use crate::catalogs::meta_backend::TableInfo;
+
+struct InMemoryTableInfo {
+    pub(crate) name2meta: HashMap<String, Arc<TableInfo>>,
```

Comment:
Perhapes here we can use InMemoryMetas in  `util/in_memory_metas.rs`?

Reply:
good idea, let me make in_memory_metas generic

Reply:
oops,  `util/in_memory_metas.rs` is not suitable to be used in `embedded_meta_backend`, the inner elements of them are quite different (TableInfo vs TableMeta), can hardly unify them without introducing new traits (or using macro), which I feel is not worthy.